ibis/backends/flink/utils.py

```diff
@@ -206,3 +210,78 @@ def format_as_string(self) -> str:
             "TO SECOND"
             f"{format_precision(self.precisions[IntervalUnit.SECOND], IntervalUnit.SECOND)}"
         )
+
+
+def _translate_interval(value, dtype):
+    """Convert interval to Flink SQL type.
+
+    Flink supports only two types of temporal intervals: day-time intervals with up to nanosecond
+    granularity or year-month intervals with up to month granularity.
+
+    An interval of year-month consists of +years-months with values ranging from -9999-11 to +9999-11.
+    An interval of day-time consists of +days hours:minutes:seconds.fractional with values ranging from
+    -999999 23:59:59.999999999 to +999999 23:59:59.999999999.
+
+    The value representation is the same for all types of resolutions.
+
+    For example, an interval of months of 50 is always represented in an interval-of-years-to-months
+    format (with default year precision): +04-02; an interval of seconds of 70 is always represented in
+    an interval-of-days-to-seconds format (with default precisions): +00 00:01:10.000000.
+    """
+    if dtype.unit in YearsToMonthsInterval.units:
+        interval = YearsToMonthsInterval(value, dtype.unit.value)
+    else:
+        interval = DaysToSecondsInterval(value, dtype.unit.value)
+
+    interval_segments = interval.interval_segments
+    nonzero_interval_segments = {k: v for k, v in interval_segments.items() if v != 0}
+
+    # YEAR, MONTH, DAY, HOUR, MINUTE, SECOND
+    if len(nonzero_interval_segments) == 1:
+        unit = next(iter(nonzero_interval_segments))
+        value = nonzero_interval_segments[unit]
+        return f"'{value}' {unit.value}{format_precision(value, unit)}"
+
+    # YEAR TO MONTH, DAY TO SECOND
+    return interval.format_as_string()
+
+
+def literal(op: ops.Literal) -> str:
```

Comment:
Can you rename this to something like `translate_literal`? The bare `literal` is easy to confuse with `ibis.literal`.

Reply:
Done!